New medication

BEKHZOD98 opened the pull request:
New Medication profile

Commit:
new_Medication

Commit:
updated

Commit:
New Medication updated

Commit:
Medication updated

Commit:
updated

vadi2 committed:
fix typos and add medication example

- Fix doseform-cs id typo (dosoform -> doseform)
- Fix medication-status alias typo
- Remove duplicate aliases
- Fix UCUM title/description typos
- Remove trailing space in ATC supplement
- Add valueset-supplement extension to volume units VS
- Add example-medication instance

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

vadi2 committed:
Merge branch 'new_Medication' of https://github.com/BEKHZOD98/digital-health-ig into new_Medication

vadi2 committed:
Merge branch 'main' into new_Medication

vadi2 committed:
Add MedicineIdentifierTypeCS and update IdentifierTypeVS

vadi2 reviewed (commented):
Need to add namingsystems for the new organizations, and create the systems according to new system structure we have (see [identifiers.md](https://github.com/uzinfocom-org/digital-health-ig/blob/main/input/pagecontent/identifiers.md)). There was an example missing as well, I started on one, but it needs to be finished.

github-actions[bot] committed:
Update DHP Capability Statement

BEKHZOD98 committed:
Merge pull request #1 from BEKHZOD98/automated/update-capability-statement

Update DHP Capability Statement (bot)

Commit:
Merge branch 'main' of https://github.com/BEKHZOD98/digital-health-ig into new_Medication

Commit:
Merge branch 'new_Medication' of https://github.com/BEKHZOD98/digital-health-ig into new_Medication

Commit:
Medication examples added

vadi2 commented on input/fsh/UZCoreMedication.fsh:
```diff
@@ -0,0 +1,184 @@
+  * use = #official
+  * value 1..1 MS
+* code MS
+* code from MedicationClassificationVS 
```
Need to be explicit about the binding strength here and in lines 76,80,88

vadi2 committed:
Sort aliases ascending

vadi2 committed:
Fix FSH review issues in Medication profile

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

Commit:
Merge branch 'main' of https://github.com/BEKHZOD98/digital-health-ig into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

Commit:
Merge branch 'new_Medication' of https://github.com/BEKHZOD98/digital-health-ig into new_Medication

Commit:
Medication excel files added

vadi2 committed:
Remove duplicated excels / csvs

Commit:
namingsystem fsh updated

Commit:
Merge branch 'new_Medication' of https://github.com/BEKHZOD98/digital-health-ig into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

BEKHZOD98 committed:
Merge branch 'main' into new_Medication

Commit:
alias

Commit:
Medication profile Ucum code ConceptMaps removed

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

github-actions[bot] committed:
Update DHP Capability Statement

BEKHZOD98 committed:
Merge branch 'main' into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

BEKHZOD98 committed:
Merge branch 'main' into new_Medication

BEKHZOD98 committed:
ATC codes updated

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into main

BEKHZOD98 committed:
Merge pull request #2 from BEKHZOD98/automated/update-capability-statement

Update DHP Capability Statement

BEKHZOD98 committed:
Merge branch 'main' into new_Medication

BEKHZOD98 committed:
CapabilityStatement

BEKHZOD98 committed:
updated

BEKHZOD98 committed:
updated ignoreWarnings

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

BEKHZOD98 committed:
updated

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

BEKHZOD98 committed:
fixed some errors

BEKHZOD98 committed:
Merge branch 'main' into new_Medication

BEKHZOD98 committed:
fixed some errors

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

vadi2 committed:
Merge remote-tracking branch 'origin/main' into new_Medication

# Conflicts:
#	input/fsh/Aliases.fsh

vadi2 committed:
remove duplicate alias block from Aliases.fsh

vadi2 commented on input/fsh/UZCoreMedication.fsh:
```diff
@@ -0,0 +1,184 @@
+    productServiceID 0..1 MS and
+    registrationCertificateID 0..1 MS
+
+* identifier[uzpharmBoxGroupID]
```
Тут нам стоит добавить ^short с текстром от NamingSystem.description, чтобы было понятней о чём identifier когда смотришь на сам профиль.

ZuhraCodes replied:
Добавили ^short из NamingSystem.description

vadi2 commented on input/fsh/UZCoreMedication.fsh:
```diff
@@ -0,0 +1,184 @@
+  * value 1..1 MS
+  */
+* code MS
+* code from MedicationClassificationVS 
```
Надo сказать силу связки здесь и в других местах.

ZuhraCodes replied:
Добавили силы связки

vadi2 committed:
enable profiled example-medication

remove orphaned identifier[uzpharmInfo] ref (slice was removed from profile)

vadi2 committed:
drop bare placeholder Medication example

vadi2 commented on input/fsh/terminology/MedicineIdentifierTypeCS.fsh:
```diff
@@ -0,0 +1,47 @@
+CodeSystem: MedicineIdentifierTypeCS
```
Смысл этой у кодовой системы есть и я его вижу! Но, 

FHIR [говорит для  Identifier.type](https://hl7.org/fhir/datatypes-definitions.html#Identifier.type):

> ... It SHOULD not be used for codes that correspond 1..1 with the Identifier.system ...

У нас тут все Identifier.type 1:1 сопоставляются с Identifier.system, так что это идёт против правила.

Стоит нам убрать эту кодовую систему и её использование в профиле.

(в общем смысле никто не сможет же использовать наши Identifier.type вне узбекистана тоже)

ZuhraCodes replied:
Исправили

vadi2 committed:
ATCClassificationCS: set supplement version to 2025, drop dead commented concepts

Version was the FHIR version (5.0.0) by mistake; tx.fhir.org serves WHO ATC as
version 2025. Also remove ~530 lines of commented-out concepts (incl. 7 codes
with Cyrillic homoglyphs) that produced nothing.

vadi2 committed:
pregenerate ATC supplement to vocabulary, move source to manual-fsh

Skips SUSHI reparse of ~5200 ATC concepts each build. Mirrors the existing
pregenerated CodeSystems (#150). Verified: SUSHI 0/0, QA unchanged (52/0/4802),
medication-classification-vs still expands (16244 concepts).

vadi2 committed:
MedicationDoseFormCS: remove 12 duplicate dose-form concepts

CRE110-118 and DIS106-108 were verbatim duplicates (same uz/ru) of
CRE101-109 and DIS102-104 in the source. 848 -> 836 concepts.

vadi2 committed:
pregenerate INN and dose-form CodeSystems, move sources to manual-fsh

Skips SUSHI reparse of ~10k concepts (INN 9347, dose-form 836) each build,
per the >100-code guideline and existing pregenerated CodeSystems (#150).
Verified: SUSHI 0/0, QA unchanged (52/0/4802), medication-doseform-vs (836)
and medication-classification-vs (16244) still expand.

vadi2 committed:
INNMedicationNameCS: remove dead commented placeholder concepts

Two commented-out sentinel concepts ("Approved INN not exists", "INN not
given") used the literal placeholder "#local code" rather than a real code.
Already absent from the pregenerated JSON; this just tidies the source.

vadi2 committed:
fix ignoreWarnings.txt botched merge

Revert this branch's additions (nested conflict markers from a 3-way merge)
to main's version. Also drops the UCUM-supplement ignore entry, currently
unused since that supplement is commented out.

vadi2 committed:
remove unused fsh.ini

Not read by SUSHI (uses sushi-config.yaml) or any build script - inert.

vadi2 commented on input/manual-fsh/MedicationDoseFormCS.fsh:
```diff
@@ -0,0 +1,2518 @@
+CodeSystem: MedicationDoseFormCS
+Id: medication-doseform-cs
+Title: "Medication dose form"
+Description: "Local code system of pharmaceutical dose forms (e.g., tablet, capsule, solution)."
```
Тут не хватает контекста, локальная это как? Кто её сделал и поддерживает?

ZuhraCodes replied:
Контекст изменено

vadi2 committed:
ATCClassificationCS: drop redundant Uzbek display from supplement concepts

Display duplicated the #uz designation and conflicted with the
supplement's #en language. Removed from all 5207 concepts in the FSH
source and regenerated the pre-generated vocabulary JSON to match.

vadi2 committed:
MedicationTotalVolume: enable UCUM units supplement, move rutherford to local CS

vadi2 committed:
UcumTo{Ru,Uz}UnitsDisplayCM: enable display ConceptMaps

vadi2 committed:
ignoreWarnings: suppress UCUM source not-present warning

vadi2 reviewed (commented):
Профиль хорош, надо сделать немного улучшений:

vadi2 committed:
INNMedicationNameCS: zero-pad code 09343 to 009343

BEKHZOD98 committed:
Merge branch 'main' into new_Medication

BEKHZOD98 committed:
Merge branch 'main' into new_Medication

ZuhraCodes committed:
fix mistakes

BEKHZOD98 committed:
Merge pull request #46 from ZuhraCodes/new_Medication

fix mistakes

BEKHZOD98 committed:
Merge branch 'uzinfocom-org:main' into new_Medication

vadi2 committed:
Add identifier slice shorts and binding strengths to UZCoreMedication

vadi2 committed:
Remove MedicineIdentifierTypeCS; Identifier.type was 1:1 with system

vadi2 committed:
Drop 'local' from MedicationDoseFormCS description

vadi2 committed:
Sync dose form CS description, drop 'local'

BEKHZOD98 committed:
Merge branch 'main' into new_Medication
